src/test/java/io/confluent/connect/jdbc/source/integration/PostgresOOMIT.java

```diff
@@ -0,0 +1,163 @@
+    props.put(JdbcSourceConnectorConfig.MODE_CONFIG, JdbcSourceConnectorConfig.MODE_BULK);
+    props.put(JdbcSourceTaskConfig.TOPIC_PREFIX_CONFIG, "topic_");
+  }
+
+  public void startTask() {
+
+    task = new JdbcSourceTask();
+    task.start(props);
+  }
+
+  @After
+  public void after() {
+    if (task != null) {
+      task.stop();
+    }
+  }
+
+  @Test(expected = OutOfMemoryError.class)
+  public void assertOutOfMemoryWithLargeBatch() throws InterruptedException {
+    props.put(JdbcSourceTaskConfig.TABLES_CONFIG, "");
+    props.put(JdbcSourceConnectorConfig.QUERY_CONFIG, LARGE_QUERY);
+    props.put(JdbcSourceConnectorConfig.BATCH_MAX_ROWS_CONFIG, Long.toString(LARGE_QUERY_ROW_COUNT));
+    startTask();
+    task.poll();
+  }
+
+  @Test
+  public void testStreamingReads() throws InterruptedException {
+    props.put(JdbcSourceTaskConfig.TABLES_CONFIG, "");
+    props.put(JdbcSourceConnectorConfig.QUERY_CONFIG, LARGE_QUERY);
+    startTask();
+    assertTrue(task.poll().size() > 0);
+  }
+
+  @Test
+  public void testTableLocksWithStreamingReads() throws InterruptedException, SQLException {
+    createTestTable();
+    props.put(JdbcSourceTaskConfig.TABLES_CONFIG, "test_table");
+    startTask();
+    assertNoLocksOpen(task);
+    assertTrue(task.poll().size() > 0);
+    assertNoLocksOpen(task);
+    task.stop();
+    assertNoLocksOpen(task);
+  }
+
+  private void createTestTable() throws SQLException {
+    log.info("Creating test table");
+    try (Connection c = pg.getEmbeddedPostgres().getPostgresDatabase().getConnection()) {
+      try (Statement s = c.createStatement()) {
+        s.execute("CREATE TABLE test_table ( c1 text )");
+        s.execute("INSERT INTO test_table VALUES ( 'Hello World' )");
+      }
+    }
+    log.info("Created table");
+  }
+
+  private void assertNoLocksOpen(JdbcSourceTask task) throws SQLException {
+    log.info("Checking for orphaned locks");
+    int count = 0;
+    try (Connection c = pg.getEmbeddedPostgres().getPostgresDatabase().getConnection()) {
+      try (Statement s = c.createStatement()) {
+        try (ResultSet rs = s.executeQuery(
+            "select * from pg_stat_activity where state like 'idle in%'"
+        )) {
+          int columnCount = rs.getMetaData().getColumnCount();
+          StringBuilder header = new StringBuilder();
+          for (int i = 1; i <= columnCount; i++) {
+            header.append(rs.getMetaData().getColumnName(i));
+            header.append("\t");
+          }
+          log.debug(header.toString());
+          while (rs.next()) {
+            count++;
+            StringBuilder row = new StringBuilder();
+            for (int i = 1; i <= columnCount; i++) {
+              row.append(rs.getObject(i));
+              row.append("\t");
+            }
+            log.debug(row.toString());
+          }
+        }
+      }
+    }
+    assertEquals("Found idle locks left open", 0, count);
+    log.info("No locks found");
```

Comment:
we usually don't have log statements in tests because they are mostly run in jenkins (no one is watching), and if there are no asserts, then it is assumed that the scenario they are testing did not error.

Reply:
This was a debugging statement for me, the assert above it captures what I want the function to verify.